content/en/docs/ops/configuration/traffic-management/dns-proxy/index.md

```diff
@@ -100,6 +100,10 @@ to [the following section](#external-tcp-services-without-vips) for more details
 
 To work around these issues, the DNS proxy additionally supports automatically allocating addresses for `ServiceEntry`s that do not explicitly define one. This is configured by the `ISTIO_META_DNS_AUTO_ALLOCATE` option.
 
+{{< tip >}}
+Please see [DNS Auto Allocation V2](/docs/ops/configuration/traffic-management/dns-proxy/#dns-auto-allocation-v2) for a new enhanced implementation of auto allocation supported by Istio from 1.23 onwards.
```

Comment:
Should we explicitly mention that `ISTIO_META_DNS_AUTO_ALLOCATE` is deprecated/not preferred, if in fact that is the case? Implying that it will be removed in N releases?

Reply:
Have modified the tip to include this, please take a look.

Reply:
Should I revert the deprecation part then? Please decide and let me know.

Reply:
I would probably vote to not mention deprecation in our documentation until that is the official decision. We should probably bring it up in the community WG I suppose. Easy enough to adjust the note later. @bleggett, would you be opposed to leaving it out until we know for sure it is our plan to remove the legacy impl?

Reply:
> 1 sort of related topic though is that there is near as makes no different no support for v1 in ambient. If you are using ambient then you need to use v2.

We should, at least, mention this explicitly here.

Reply:
> I'm fine waiting on a deprecation intent decision - but in general I don't think we should be documenting features that exactly and fully replace other features (as this does AFAICT) if we don't at least articulate an opinion on what we _plan_ to do with the old one to users.
> 
> "We have two things that do exactly the same thing. Here they are." is not ideal messaging, and provides no guidance.

In this case then how about replacing the old content with the new flag, and not mention anything about the old flag?

Reply:
> In this case then how about replacing the old content with the new flag, and not mention anything about the old flag?

Probably too confusing/disruptive to do that until we have confirmation the old one is going to be deprecated (and we actually deprecate it), like @ilrudie said.

```
Please see [DNS Auto Allocation V2](/docs/ops/configuration/traffic-management/dns-proxy/#dns-auto-allocation-v2) for an enhanced implementation of auto allocation supported by Istio from 1.23 onwards. This enhanced implementation is recommended for sidecar mode and required for ambient mode.
```

Reply:
@bleggett suggested wording reads nicely to me. big +1

Reply:
@bleggett : Done